ADR/0014-let-pipelines-proceed.md

```diff
@@ -0,0 +1,53 @@
+# 14. Let Pipelines Proceed
+
+* Date Decided: 2022-05-??
+* Date Documented: 2023-01-31
+
+## Status
+
+Accepted
+
+Relates to [ADR 13. Stonesoup Test Stream - API contracts](0013-integration-service-api-contracts.html)
+
+## Context
+
+The user's build pipeline includes scanning and linting tasks that operate on the source code and
+the built image (SAST, antivirus, clair, etc..). The purpose of these tasks is to find problems in
+the user's source code or dependencies and alert the user so they can take action ([STONE-459]).
+
+One frustration we've heard from users in previous systems is that they don't want to be blocked in
+their development or testing by complications in the build system, by compliance concerns. We want
+to fix that by offering a system that permits as much progress in the lifecycle of the user's
+application (build, test, and pre-production deployment) but which also protects production from
+non-compliant builds via mechanisms in the [enterprise contract].
+
+A problem we face: in Tekton, a failing TaskRun causes the whole PipelineRun to fail. If the purpose
+of our linting and scanning tasks is to find problems - which usually looks like failure with
+a non-zero exit code - how do they do their job without constantly breaking the user's build,
+stopping them from running integration tests, and stopping them from deploying candidate builds to
+their lower [Environments]?
+
+## Decision
+
+All scanning and linting TaskRuns should *succeed* even if they find problems in the content they
+are evaluating.
+
+Use the `HACBS_TEST_OUTPUT` result convention from [ADR-0013] to expose those results and render them
+for users ([STONE-459]).
+
+## Consequences
+
+* Users should find that even if their scanners find problems, they can still build, test, and
+  deploy to lower Environments.
```

Comment:
What is considered a `lower Environment` here?

Reply:
Left intentionally vague for the purposes of the ADR. Anything that is "not production". Here, we're still assuming that the user uses the managed-workspace release model via the release service.